Tasks/SCP/task.json

```diff
@@ -0,0 +1,113 @@
+  ],
+  "inputs": [
+    {
+      "name": "sshEndpoint",
+      "type": "connectedService:ssh",
+      "label": "SSH endpoint",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "SSH endpoint with connection details for the remote machine."
+    },
+    {
+      "name": "sourceFolder",
+      "type": "filePath",
+      "label": "Source folder",
+      "defaultValue": "",
+      "required": false,
+      "helpMarkDown": "The source folder that the copy pattern(s) will be run from.  Empty is the root of the repo.  Use [variables](https://go.microsoft.com/fwlink/?LinkID=550988) if files are not in the repo. Example: $(agent.builddirectory)"
+    },
+    {
+      "name": "contents",
+      "type": "multiLine",
+      "label": "Contents",
+      "defaultValue": "**",
+      "required": true,
+      "helpMarkDown": "File paths to include as part of the copy. Supports multiple lines of minimatch patterns. [More Information](https://go.microsoft.com/fwlink/?LinkID=708389)"
+    },
+    {
+      "name": "targetFolder",
+      "type": "string",
+      "label": "Target folder",
+      "defaultValue": "",
+      "required": true,
+      "helpMarkDown": "Target folder on remote machine where files will be copied to. Example: /home/user/MySite"
+    },
+    {
+      "name": "cleanTargetFolder",
+      "type": "boolean",
+      "label": "Clean target folder",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Delete all existing files in target folder before copy",
+      "groupName": "advanced"
+    },
+    {
+      "name": "overwrite",
+      "type": "boolean",
+      "label": "Overwrite",
+      "defaultValue": "true",
+      "required": false,
+      "helpMarkDown": "Replace existing files in target folder",
+      "groupName": "advanced"
+    },
+    {
+      "name": "flattenFolders",
+      "type": "boolean",
+      "label": "Flatten folders",
+      "defaultValue": "false",
+      "required": false,
+      "helpMarkDown": "Flatten the folder structure and copy all files into specified target folder on the remote machine.",
+      "groupName": "advanced"
+    }
+  ],
+  "execution": {
+    "Node": {
+      "target": "scp.js",
+      "argumentFormat": ""
+    }
+  },
+  "messages": {
+    "UseDefaultPort": "Using port 22 which is the default for SSH since no port was specified.",
+    "SourceNotFolder": "Source folder has to be a valid folder path.",
+    "CleanTargetFolder": "Cleaning target folder %s on the remote machine",
+    "CleanTargetFolderFailed": "Failed to clean the target folder on the remote machine. %s",
+    "NothingToCopy": "No files were found matching the patterns specified to copy to the remote machine.",
+    "CopyingFiles": "Found %s files to copy to the remote machine.",
+    "StartedFileCopy": "Copying file %s to %s on remote machine.",
+    "CopyCompleted": "Completed copying %s files to the remote machine.",
+    "FileExists": "File %s already exists on the remote machine.",
```

Comment:
P2: "File %s cannot be copied to the remote machine because it already exists and the 'Overwrite' option is disabled."